### ISSUE TITLE
Notify user of accepted or rejected changes

### ISSUE THREAD

**Issue by doomspork** [beam-community/elixir-companies#454]
We need a way to tell users who've submitted a change that we accepted it or rejected it and why.

**Comment by doomspork** [comment]
We could capture the user's email address from GH and email them a notification?

**Comment by gemantzu** [comment]
I think I like the first solution better. The other way around would be to show users the pending changes they have on the page. We may want to have an option to ask them for changes for something instead of only just reject.

**Comment by zacck** [comment]
I'd like to handle this,  Perhaps we can use something like Bamboo and send out an email whenever these approvals or rejections occur? We would need something like a Sendgrid apikey to use, and  some agreed form of template to use when sending out this email ... should be kinda straight forward

**Comment by doomspork** [comment]
@zacck that was going to be my plan of attack! I will create a Sendgrid account for this project (if I haven't already 🤔 ). We can use Sendgrid templates themselves directly and keep the email in the app pretty straight forward.  This is what I'm doing in another application:

```elixir
defmodule Notify.Email do
  import Bamboo.Email
  import Bamboo.SendGridHelper

  def forgot_pass(%{email: email, first_name: first_name, forgot_pass_code: code}) do
    new_email()
    |> to(email)
    |> from("...")
    |> subject("...")
    |> with_template("<SENDGRID_TEMPLATE_ID>") # easily configurable
    |> add_dynamic_field(:first_name, first_name) # whatever fields we'd need
    |> add_dynamic_field(:code, code)
  end
end
```

**Comment by zacck** [comment]
@doomspork  haha yes exactly ... now to find those approval/rejection events where are they exactly?

**Comment by doomspork** [comment]
@zacck they don't exist yet, I was going to do them along with the rest of this. Let me take a peek at some things and I can put together a branch we can share to tackle this 😁

**Comment by zacck** [comment]
@doomspork  I went spelunking I found them thoug
I feel as though we need to notify the change owner at this point 
https://github.com/doomspork/elixir-companies/blob/8c0d12772e24aca554e80041080e8f0c2335adb0/lib/companies/pending_changes.ex#L145

or am I wrong?

**Comment by zacck** [comment]
You could make two templates one for approval and one for rejection and depending on the `approval` param I could send out a message to the owner of the pending change?

**Comment by doomspork** [comment]
Here is where we kick off the notifications for new changes: https://github.com/doomspork/elixir-companies/blob/8c0d12772e24aca554e80041080e8f0c2335adb0/lib/companies/pending_changes.ex#L125

Right now the `Notify` code only supports a single notifier at a time so we'll need to make some adaptions for this.

**Comment by doomspork** [comment]
This is just a quick stab at it, I can take a more in-depth look later. To achieve what we need we can probably do something like this:

```elixir
defmodule Notify do
  alias Notify.Email

  def perform(%{approved: true} = pending_change), do: Email.change_approved(pending_change)
  def perform(%{approved: false} = pending_change), do: Email.change_rejected(pending_change)
  def perform(message), do: apply(notifier(), :notify, [message])

  defp notifier, do: Application.get_env(:companies, :notifier)
end
```

@zacck I'll create the templates but we won't want to include them in the code, they should be loaded from the `System`:

```elixir
System.get_env("SENDGRID_APPROVED_TEMPLATE_ID")
System.get_env("SENDGRID_REJECTED_TEMPLATE_ID")
```

In the email template we'll need at least 4 (5?) dynamic fields:

- Github handle of the user ("Hello doomspork :wave:, ")
- Resource type. Did they change a `company` or a `job`, so the verbiage can reflect this
- The resource name (company name or job title)
- Note from the approver
- Github handle of the approver (this _could_ be included in the above message: `" ... " — doomspork`

Thoughts? What I don't _think_ you'll need anything from me WRT to Sendgrid. I will warn you, haven't done this just recently, I don't believe Bamboo's `master` works with the latest Sendgrid template API.

**Comment by zacck** [comment]
@doomspork yeah good catch we run into this at work routinely, you need to build legacy templates, your other details and the code sample seems to be what I would have done exactly will set some time aside to push this in

**Comment by zacck** [comment]
@doomspork  Started on this and kinda got stuck with finding the actual resource a change belongs to since I need the title or name in the template any ideas on how to do this?

**Comment by zacck** [comment]
@doomspork  👀  watching you

**Comment by doomspork** [comment]
Responded on the PR 😁

**Comment by gemantzu** [comment]
solved by #550!
